packages/components/src/components/ScalarCopyButton/ScalarCopyButton.vue

```diff
@@ -0,0 +1,101 @@
+<script setup lang="ts">
+import { ScalarIconCheck, ScalarIconCopy } from '@scalar/icons'
+import { cva, cx } from '@scalar/use-hooks/useBindCx'
+import { useClipboard } from '@scalar/use-hooks/useClipboard'
+import { computed, ref } from 'vue'
+
+import { ScalarButton } from '../ScalarButton'
+
+const props = withDefaults(
+  defineProps<{
+    /** Content to copy to clipboard */
+    content: string | object
+    /** Optional aria label for accessibility */
+    ariaLabel?: string
+    /** Size variant */
+    size?: 'sm' | 'md'
+  }>(),
+  {
+    ariaLabel: 'Copy content',
+    size: 'md',
+  },
+)
+
+const { copyToClipboard } = useClipboard()
+const showCopied = ref(false)
+
+const handleCopy = async () => {
+  // Prevent multiple clicks while already copying
+  if (showCopied.value) {
+    return
+  }
+
+  const contentToCopy =
+    typeof props.content === 'string'
+      ? props.content
+      : JSON.stringify(props.content, null, 2)
+
+  await copyToClipboard(contentToCopy)
+  showCopied.value = true
+
+  setTimeout(() => {
+    showCopied.value = false
+  }, 1200)
+}
+
+const variants = cva({
+  base: 'absolute top-0 right-0 h-8 w-8 p-0 flex items-center justify-center brightness-lifted bg-inherit rounded focus-visible:opacity-100 -outline-offset-1',
+  variants: {
+    showCopied: {
+      true: 'text-c-1',
+      false: 'text-c-3 hover:text-c-1',
+    },
+  },
+})
+</script>
+
+<template>
+  <div
+    class="scalar-code-copy opacity-0 sticky flex inset-0 justify-end items-start bg-inherit">
+    <ScalarButton
+      :class="cx(variants({ showCopied }))"
+      type="button"
+      variant="ghost"
+      @click="handleCopy">
+      <span class="sr-only">{{ ariaLabel }}</span>
+      <ScalarIconCopy
+        class="copy-icon size-4"
+        :class="{ copied: showCopied }" />
+      <ScalarIconCheck
+        class="check-icon size-4"
+        :class="{ visible: showCopied }" />
+    </ScalarButton>
+  </div>
+</template>
+
+<style scoped>
+.scalar-code-block:hover .scalar-code-copy {
```

Comment:
This should probably be up in the `ScalarCodeBlock` component

Reply:
yes not against it!